### PR TITLE
Vendoring tasks

### LINKED ISSUE #2474: task for vendoring pluggy
Next time we vendor pluggy, we should create an `invoke` task to do that automatically; also, the same task should update all internal links in the docs to point to the new location, for example:

```rst
same arguments as ``before`` but also a :py:class:`_CallOutcome`` object
```

To:

```rst
same arguments as ``before`` but also a :py:class:`_CallOutcome <_pytest.vendored_packages.pluggy._CallOutcome>` object
```

### PR DESCRIPTION
**Pull request by RonnyPfannschmidt**
fixes #2474

### CODE REVIEW

File: changelog/2474.trivial [hunk 1]
```diff
@@ -0,0 +1 @@
+create invoke tasks for updating the vendored packages
```
Comment (nicoddemus) [inline changelog/2474.trivial]: Please use proper grammar here. 😉 

```
Create invoke tasks for updating the vendored packages.
```

File: tasks/vendoring.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+def remove_libs(ctx):
+    print("removing vendored libs")
+    for path in VENDOR_TARGET.listdir():
+        if path.basename not in GOOD_FILES:
+            print(" ", path)
+            path.remove()
+
+@invoke.task(pre=[remove_libs])
+def update_libs(ctx):
+    print("installing libs")
```
Comment (nicoddemus) [inline tasks/vendoring.py]: I think it makes sense to call `remove_libs` automatically first.

File: tasks/vendoring.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    ctx.run("pip install -t {target} pluggy".format(target=VENDOR_TARGET))
+    ctx.run("git add {target}".format(target=VENDOR_TARGET))
+    print("please commit to finish the updae after running the tests")
```
Comment (nicoddemus) [inline tasks/vendoring.py]: I suggest also outputting a suggestion for the commit message, so it is standardized:

```python
print("Please commit to finish the updae after running the tests:") 
print()
print('    git commit -am "Updated vendored libs"')
```

File: tasks/vendoring.py (hunk 2 above)
Comment (RonnyPfannschmidt) [inline tasks/vendoring.py]: it is ^^  - as pre task

Comment (nicoddemus) [inline tasks/vendoring.py]: Oops completely missed that. 😅

**Comment by nicoddemus** [comment]
Thanks! 👍

**Pull request merged**